### DISCUSSION TITLE
Protect the entire wp_options table all at once

### DISCUSSION THREAD

**Discussion by jdy98p** [snicco/fortress#2] [Feature Requests]
Make Vaults&Pillars easy to protect the **entire** wp_options table all at once (not single entries), with new entries being auto-protected so I don't even have to think about new plugins adding options I need to add protection to.

That should increase speed and security the easy, "lazy" way.

**Comment by calvinalkan** [comment]
I've looked at this a bit, this is only possible by reducing the available security for not explicitly defined options. 

For example, right now, we throw an error if a defined option does NOT exist in the database. 

That could only be mainteded if effectivly no more options can be added. 

It's not as clearcut, but none the less a good idea. 

Will keep this in mind to see if it's feasible.

**Reply by Mte90** [reply]
So there is a filter to read all the WP options so indeed can be turned on in all of them but I guess that is too much expensive for the website for all of them, anyway with a DB cache layer the problem can be another issue if you want that is just one time decryption.

**Reply by calvinalkan** [reply]
It's complicated, lots of moving parts in that one. 

But I'm not discarding the idea (yet)
